waiter/src/waiter/token.clj

```diff
@@ -434,7 +434,8 @@
       (when (not-empty unknown-keys)
         (throw (ex-info (str "Unsupported key(s) in token: " (str (vec unknown-keys)))
                         {:status http-400-bad-request :token token :log-level :warn}))))
-    (let [service-parameter-with-token-defaults (attach-token-defaults-fn new-service-parameter-template)]
+    (let [service-parameter-with-token-defaults (attach-service-defaults-fn new-service-parameter-template)
+          missing-parameters (->> sd/service-required-keys (remove #(contains? new-service-parameter-template %1)) seq)]
```

Comment:
Why is `missing-parameters` defined in terms of `new-service-parameter-template` and not `service-parameter-with-token-defaults`?

Reply:
Oops, copy-paste error. Fixed and added assertions in tests to validate the reported missing parameters.